packages/front-end/components/Features/FeaturesStats.tsx

```diff
@@ -0,0 +1,148 @@
+import { useRouter } from "next/router";
+import { useMemo } from "react";
+import { FeatureCodeRefsInterface } from "back-end/types/code-refs";
+import { OrganizationSettings } from "back-end/types/organization";
+import { FaGitAlt, FaExternalLinkAlt } from "react-icons/fa";
+import Code from "@/components/SyntaxHighlighting/Code";
+import { useUser } from "@/services/UserContext";
+import Button from "../Button";
+import PremiumTooltip from "../Marketing/PremiumTooltip";
+
+const generatePlatformUrl = (
+  platformUrl: string,
+  repo: string,
+  branch: string,
+  filePath: string,
+  lineNumber: number
+) => {
+  return `${platformUrl}/${repo}/blob/${branch}/${filePath}#L${lineNumber}`;
+};
+
+export default function FeaturesStats({
+  orgSettings,
+  codeRefs: allCodeRefs,
+}: {
+  orgSettings: OrganizationSettings;
+  codeRefs: FeatureCodeRefsInterface[];
+}) {
+  const router = useRouter();
+  const {
+    codeReferencesEnabled,
+    codeRefsPlatformUrl,
+    codeRefsBranchesToFilter,
+  } = orgSettings;
+  const { hasCommercialFeature } = useUser();
+  const hasFeature = hasCommercialFeature("code-references");
+
+  const codeRefs = useMemo(() => {
+    if (!codeRefsBranchesToFilter || codeRefsBranchesToFilter.length === 0) {
+      return allCodeRefs;
+    }
+    return allCodeRefs.filter((codeRef) =>
+      codeRefsBranchesToFilter.includes(codeRef.branch)
+    );
+  }, [allCodeRefs, codeRefsBranchesToFilter]);
+
+  if (!codeReferencesEnabled) {
+    return (
+      <div className="contents container-fluid pagecontents">
+        <div
+          className="appbox bg-white"
+          style={{
+            height: "18rem",
+            display: "flex",
+            flexDirection: "column",
+            justifyContent: "center",
+            alignItems: "center",
+          }}
+        >
+          <h2>Enable Code References</h2>
+          <p style={{ width: "32rem" }}>
+            Quickly see instances of feature flags being leveraged in your
+            codebase, with direct links from GrowthBook to the platform of your
+            choice.
+          </p>
+          {hasFeature ? (
+            <Button
+              onClick={async () => {
+                router.push("/settings#configure-code-refs");
+              }}
+            >
+              Go to settings
+            </Button>
+          ) : (
+            <PremiumTooltip commercialFeature="code-references">
+              <Button
+                color="warning"
+                onClick={async () => {
+                  router.push("/settings");
+                }}
+              >
+                Upgrade to Pro
+              </Button>
+            </PremiumTooltip>
```

Comment:
Small nit: I don't think routing users to the `/settings` page is the right destination here - the user can't upgrade their plan from that page.

Instead, we have the `UpgradeModal` which has some permission logic built into it. We can also differentiate the button CTA based on the plan. E.G. if the org is `oss` we can prompt them to `Try Enterprise` vs `Upgrade to Pro`.

```
  <button
                    className="btn btn-premium font-weight-normal"
                    onClick={() => setUpgradeModal(true)}
                  >
                    {accountPlan === "oss" ? (
                      <>
                        Try Enterprise <GBPremiumBadge />
                      </>
                    ) : (
                      <>
                        Try Pro <GBPremiumBadge />
                      </>
                    )}
                  </button>
```

[Here is a reference](https://github.com/growthbook/growthbook/blob/main/packages/front-end/components/License/ShowLicenseInfo.tsx) for how we do this on the SideBar.